modules/calib3d/test/test_solvepnp_ransac.cpp

```diff
@@ -837,6 +837,92 @@ TEST(Calib3d_SolvePnPRansac, double_support)
+        { -3.28314781e-01, -3.30749929e-01 }, { -3.46136659e-01, -1.49928510e-01 },
+        { -3.50446790e-01, -1.42250136e-01 }, { -4.83760357e-01, -6.05802834e-02 },
+        { -5.34562290e-01, -4.90532629e-02 }, { -4.88979757e-01, -6.34466186e-02 },
+        { -4.48474199e-01, -1.04112938e-01 }, { -4.21923876e-01, -1.10946946e-01 },
+        { -2.85925508e-01, -1.75974697e-01 }, { -5.09869874e-01, -6.05546944e-02 },
+        { -4.97992635e-01, -7.01366886e-02 }, { -5.05892992e-01, -8.00956413e-02 },
+        { -5.43107152e-01, -4.90474701e-02 }, { -1.39427155e-01, 6.03941306e-02 },
+        { -3.74340236e-01, -1.11043371e-01 }, { -2.89478928e-01, -1.86347634e-01 },
+        { -5.38358629e-01, -5.09638414e-02 }, { -4.88979757e-01, -6.34466186e-02 },
+        { -2.91625351e-01, -1.87459111e-01 }, { -5.04870534e-01, -1.99246198e-01 },
+        { -3.43228579e-01, -1.67184278e-01 }, { -3.40847254e-01, -1.66233093e-01 },
+        { -3.00223082e-01, -1.87433019e-01 }, { -3.54677111e-01, -1.64277285e-01 },
+        { -3.55155081e-01, -1.63796946e-01 }, { -3.92849147e-01, -2.80838192e-01 },
+        { -4.50463027e-01, -9.89229158e-02 }, { -3.14653307e-01, -1.49531603e-01 },
+        { -2.73969293e-01, -1.78405225e-01 }, { -1.09586723e-01, -2.56693006e-01 },
+        { -5.43504596e-01, -4.61776406e-02 }, { -5.59334219e-01, -4.01909500e-02 },
+        { -3.37530792e-01, -1.58576757e-01 }, { -5.54325163e-01, -1.86715424e-01 },
+        { -4.88430500e-01, -1.15605585e-01 }, { -5.40505290e-01, -2.03943357e-01 },
+        { -5.37568033e-01, -5.01672402e-02 }, { -4.21460837e-01, -2.66400576e-01 },
+        { -5.37177384e-01, -2.03950211e-01 }, { -3.13457936e-01, -1.50333717e-01 },
+        { -1.37509882e-01, 4.98050973e-02 }, { -4.02904212e-01, -1.08589850e-01 },
+        { -5.68608880e-01, -1.75223738e-01 }, { -1.04737304e-01, -1.27359554e-01 },
+        { -2.53603160e-01, -1.92443535e-01 }, { -5.14613926e-01, -6.34208024e-02 },
+        { -5.57353795e-01, -4.37783562e-02 }, { -5.55796064e-02, -2.00841814e-01 },
+        { -3.69092196e-01, -1.15367927e-01 }, { -5.43109000e-01, -4.61778566e-02 },
+        { -4.11464423e-01, -2.67393112e-01 }, { -3.55155081e-01, -1.63796946e-01 },
+        { -5.57155788e-01, -1.99532032e-01 }, { -4.80420530e-01, -7.49433413e-02 },
+        { -5.43107152e-01, -4.90474701e-02 }, { -5.54799795e-01, -1.87192410e-01 },
+        { -4.34312940e-01, -2.64443249e-01 }, { -4.76575375e-01, -1.06057212e-01 },
+        { -2.62239158e-01, -3.79666299e-01 }, { -5.07591426e-01, -2.19547823e-01 },
+        { 1.66453179e-02, -3.65559101e-01 }, { -4.36202317e-01, -2.45646402e-01 },
+        { -5.38358629e-01, -5.09638414e-02 }, { -3.46136659e-01, -1.49928510e-01 }
+    };
+
+    Mat camera_mat = Mat::eye(3, 3, CV_64FC1);
+    Mat rvec, tvec;
+    vector<uchar> inliers;
+    bool result = solvePnPRansac(pts3d, pts2d, Mat::eye(3, 3, CV_64FC1), noArray(), rvec, tvec, false, 100, 4. / 460.0, 0.99, inliers);
```

Comment:
> 4. / 460.0

Reprojection error is in pixel unit (default value is 8). Wondering if it explains why you only have a consensus set of 5 points?

Reply:
In my test data, the 2D points are from the normalized image coordiante system, that's why the camera matrix is `Mat::eye(3, 3, CV_64FC1)`. So, the reprojection error here is not in pixel unit (`1.0/f` pixel in fact).

As you can see, the focal length of my camera is about 460.0, and the tolerance of reprojection error is set to 4 pixels. It should be fine.

Reply:
You are right. I did not see that the camera matrix is identity.